gateway-service-impl/src/main/java/org/hypertrace/gateway/service/trace/TracesService.java

```diff
@@ -172,51 +172,55 @@ List<Trace> filterTraces(
   }
 
   int getTotalFilteredTraces(RequestContext context, TracesRequest request, TraceScope scope) {
-    int total = 0;
-    Builder queryBuilder = createQueryWithFilter(request, scope, context);
-    // validated that the selection is not empty
-    if (request.getSelectionCount() < 1) {
-      throw new IllegalArgumentException("Query request does not have any selection");
-    }
-
-    String firstSelectionAttributeId =
-        ExpressionReader.getAttributeIdFromAttributeSelection(request.getSelection(0))
-            .orElseThrow();
-    QueryRequest queryRequest =
-        queryBuilder
-            .addSelection(createCountByColumnSelection(firstSelectionAttributeId))
-            .setLimit(1)
-            .build();
-
-    Iterator<ResultSetChunk> resultSetChunkIterator =
-        queryServiceClient.executeQuery(context, queryRequest);
-    while (resultSetChunkIterator.hasNext()) {
-      ResultSetChunk chunk = resultSetChunkIterator.next();
-      if (LOG.isDebugEnabled()) {
-        LOG.debug("Received chunk: " + chunk.toString());
+    int total = -1;
+    if (request.getFetchTotal()) {
```

Comment:
Can we move the check to the caller instead calling the `getTotalFilteredTraces`?

Reply:
To keep the api compatible, 
- we will first merge API change as follow
- change the consumer
- provide new api implementation